Upgrade helper for bare native projects (#24773)
expo/expo: docs/pages/bare/upgrade.mdx

---
title: Upgrade your bare project
description: View file-by-file diffs of all the changes you need to make to your native projects to upgrade them to the next Expo SDK version.

> sidebar_title: Upgrading your bare project

Reviewer: ```suggestion
sidebar_title: Upgrading your bare project
hideTOC: true
```

Reviewer: alternatively: what if the sidebar showed files that are included in the diff? might be tricky, can't recall how that's implemented

> sidebar_title: Upgrading your bare project

Reviewer 2: ```suggestion
```

I don't think we need to add a sidebar title that uses Gerund. Lately I've been cleaning this up (WIP) from our docs pages. I'd suggest to remove this.

<img width="260" alt="CleanShot 2023-10-25 at 15 40 58@2x" src="https://github.com/expo/expo/assets/10234615/4545ab27-90d8-44cb-9246-128b6ab65467">

---

import { TemplateBareMinimumDiffViewer} from '~/ui/components/TemplateBareMinimumDiffViewer';

> If you manage your own native projects (also known as bare workflow), when [upgrading to the latest Expo SDK](/workflow//upgrading-expo-sdk-walkthrough/), you'll need to make some changes to your native projects. Use the diffs here to compare native project files beteween your current SDK version and your desired target SDK version, so you can make the changes needed to match the template used by your version of the `expo` package.

Reviewer: ```suggestion
If you manage your own native projects (also known as bare workflow), when [upgrading to the latest Expo SDK](/workflow/upgrading-expo-sdk-walkthrough/), you'll typically need to make some changes to your native projects. Use the diffs here to compare native project files between your current SDK version and your desired target SDK version, so you can make the changes needed to match the template used by your version of the `expo` package. The tools on this page is similar to [React Native Upgrade Helper](https://react-native-community.github.io/upgrade-helper/), but it is oriented around projects that use Expo modules and related tooling.
```

> If you manage your own native projects (also known as bare workflow), when [upgrading to the latest Expo SDK](/workflow/upgrading-expo-sdk-walkthrough/), you'll typically need to make some changes to your native projects. Use the diffs here to compare native project files between your current SDK version and your desired target SDK version, so you can make the changes needed to match the template used by your version of the `expo` package. The tools on this page is similar to [React Native Upgrade Helper](https://react-native-community.github.io/upgrade-helper/), but it is oriented around projects that use Expo modules and related tooling.

Reviewer 2: ```suggestion
If you manage your native projects (also known as bare workflow), to [upgrade to the latest Expo SDK](/workflow/upgrading-expo-sdk-walkthrough/), you have to make changes to your native projects. It can be a complex process to find which native file changes and what to update in which file.

The following guide provides diffs to compare native project files between your project's current SDK version and the target SDK version you want to upgrade. You can use them to make changes to your project depending on the `expo` package version your project uses. The tools on this page are similar to [React Native Upgrade Helper](https://react-native-community.github.io/upgrade-helper/). However, they are oriented around projects that use Expo modules and related tooling.
```

Let's expand a little on what problem this page is trying to solve. I've split it into two paragraphs. Also, I've made some modifications to make the text a bit earlier to read.

> ## Upgrade your native project files

Reviewer 2: ```suggestion
## Upgrade native project files
```

Once you have [upgraded your Expo SDK version and related dependencies](/workflow//upgrading-expo-sdk-walkthrough/#how-to-upgrade-to-the-latest-sdk-version), you can use this diff tool to see what changes you need to make to your native projects to bring them up to date with your Expo SDK version.

> Choose your from SDK version and to SDK version to see the diff. Then apply those changes to your native projects.

Reviewer: ```suggestion
Choose your from SDK version and to SDK version to see the diff. Then apply those changes to your native projects by copying and pasting or manually making changes as needed.
```

> Once you have [upgraded your Expo SDK version and related dependencies](/workflow//upgrading-expo-sdk-walkthrough/#how-to-upgrade-to-the-latest-sdk-version), you can use this diff tool to see what changes you need to make to your native projects to bring them up to date with your Expo SDK version.

Reviewer 2: ```suggestion
Once you have [upgraded your Expo SDK version and related dependencies](/workflow//upgrading-expo-sdk-walkthrough/#how-to-upgrade-to-the-latest-sdk-version), use the diff tool below to learn about changes you need to make to your native project and bring them up to date with the current Expo SDK version.
```

> Choose your from SDK version and to SDK version to see the diff. Then apply those changes to your native projects by copying and pasting or manually making changes as needed.

Reviewer 2: ```suggestion
Choose your **from SDK version** and a **to SDK version** to see the diff. Then, apply those changes to your native projects by copying and pasting or manually making changes to the project files.
```
